### ISSUE TITLE
Dark Mode Toggle Placement

### ISSUE THREAD

**Issue by kjellr** [WordPress/twentytwentyone-dark-mode#19]
I'd like revisit discussion about the toggle placement from #1. Currently, the dark mode toggle is placed here: 

![image](https://user-images.githubusercontent.com/1202812/97740888-4478bc80-1ab8-11eb-896b-da98039f3ec9.png)

I have a couple concerns with its current placement: 

## Prominence of the Button

This is a very strong visual element, and it is located above all other site content. Unless there's a prominent site logo, this button will likely be the first thing a user's eye is drawn to in the header. From a hierarchy perspective, I don't think that makes much sense: while we want to make this toggle feature available, it shouldn't be more important visually than the site title. 

This is especially an issue on mobile, where it takes up a larger proportion of the screen: 

![Screen Shot 2020-10-30 at 2 25 05 PM](https://user-images.githubusercontent.com/1202812/97743216-b9012a80-1abb-11eb-95ad-9a7e2c355cb7.png)

Does this need to exist in the header? I do not have data around this, but I'd guess that most users will not make use of this button at all. Does this need to be in the main header if that's the case?

## Position of the Button

Assuming this button does remain in the header, the current placement feels imbalanced. The header forms a weird "N" shape: 

![image](https://user-images.githubusercontent.com/1202812/97741977-0381a780-1aba-11eb-9e5c-35e1eec3ba2f.png)

If we do place the button in the header, I'd suggest we work to balance this shape out visually. Even something like this helps: 

![image](https://user-images.githubusercontent.com/1202812/97742094-362ba000-1aba-11eb-8eca-881ddbcbec89.png)

Moving it below the menu there would also improve the mobile situation, allowing the close button to once again appear in the same position as the open menu button. Currently, [they are in different spots](https://cloudup.com/cp1h9lIA1qE). 

It would be even more balanced to do something like this (without placing it in the actual `nav` element of course): 

![image2](https://user-images.githubusercontent.com/1202812/97742165-4a6f9d00-1aba-11eb-816e-6de12b4fae9a.png)

Are there concerns with this approach? Does anyone have any other ideas that I've missed?

**Comment by kjellr** [comment]
It's possible that this is just a poor option for accessibility/tab order reasons, but for the sake of throwing out ideas: have we discussed the option of having the button fixed somewhere? 

![Screen Shot 2020-10-30 at 2 39 12 PM](https://user-images.githubusercontent.com/1202812/97744671-08e0f100-1abe-11eb-98db-638603a30cb7.png)

And/or placed closer to the thumb on mobile? Possibly having it fade out on scroll? 

![Screen Shot 2020-10-30 at 2 34 41 PM](https://user-images.githubusercontent.com/1202812/97744869-578e8b00-1abe-11eb-9ec7-44d90bc0c0d2.png)

**Comment by carolinan** [comment]
I don't think the button should have a solid dark background when dark mode is off, that is why it is more distracting on small screens, compared to the menu.
Or, give the menu button the same style.

The N shape does not look strange to me except there could be some more space between the menu and the button.

For me having a full width button at the footer is more distracting than having a smaller button in the header.

**Comment by aristath** [comment]
> have we discussed the option of having the button fixed somewhere?

That is how it was until a few days ago... And the code is actually still there. It's easy to test by changing this line from `relative` to `fixed-bottom`: https://github.com/WordPress/twentytwentyone-dark-mode/blob/e8c28e8252e651a4ee93a17be1fcabbc5cc3c8c2/functions.php#L231
As a personal preference I find that the bottom-fixed option is not so much "in your face" and it becomes a secondary element in the page instead of a primary, "important" option.

Regarding the colouring scheme, we could invert the buttons... That should make them blend-in a bit nicer.

**Comment by aristath** [comment]
Moving the button to be visually after the menu is not currently possible while in plugin form (there's no hook in the theme for that). But once this gets merged in the theme later today I can definitely put it as the last item on the menu (visually, not in the DOM inside the actual `nav` element). And yes, that makes it a lot more balanced.

**Comment by kjellr** [comment]
> As a personal preference I find that the bottom-fixed option is not so much "in your face" and it becomes a secondary element in the page instead of a primary, "important" option.

Just for clarification, are you saying you think it's more appropriate for this to be a secondary element at the bottom?

> Moving the button to be visually after the menu is not currently possible while in plugin form (there's no hook in the theme for that). But once this gets merged in the theme later today I can definitely put it as the last item on the menu (visually, not in the DOM inside the actual nav element). And yes, that makes it a lot more balanced.

That sounds good, let's give that a try (as well as maybe the fixed option?) and see what it's like.

**Comment by carolinan** [comment]
Even if it is not technically inside the menu, the focus order would still need to match what we see:
logo, title, menu, dark mode button

**Comment by aristath** [comment]
> Just for clarification, are you saying you think it's more appropriate for this to be a secondary element at the bottom?

Yes. Switching color modes is a nice feature, but shouldn't interfere with the header. Switching dark/light modes is something that can be seen more as a nicety and definitely not as important as the site-navigation.
The color-scheme switch basically says _"hey there! In case your lighting conditions, personal preferences or a migraine make it hard to read the site the way you see it now, try switching to the other mode, see if that feels better for you"_.

[Discussed on Slack](https://wordpress.slack.com/archives/C02RP4VMP/p1604320591432600) and it was decided to put this on the bottom-right for the purposes of beta. Done in https://github.com/WordPress/twentytwentyone/pull/773/commits/3ed0b230cb06b2b9fd0640a952314c433d7391db
